crates/rpc/src/v03/method/simulate_transaction.rs

```diff
@@ -0,0 +1,500 @@
a2fc27a583699a3b4df200dd37a1bd9f96b55873d1f0fac756c600fff610d5b4996cc1dbfe37413febc6d0de1f313bb4bd03ad75ae7cdb8bde88871a02c8b82cfe838db62bf95b7fc9e2a401f6f1ae5149976c7ed11ca867951988c13287869c5d2b599947d2f8facf899291e2567d57bedc73fef03afd18f1db185a68db7bf67d09903111b8d111132af34c1d66148daa3bfc546dd03b4f906c2443c5fd9772fc3be5f3d10fb593dad660945f9aff14edcd2fda59006043f57a22af6d8700d43f600784df3f0157fd8abb96b9fed6000a9a32f680f30c74e21a81d811866b2a67f80da848e720533b00fdd7e3606e8037bd06906359df22facaa70d9a64c60caf043e2a620e15e3c6cff340c3d3daeed54c50288e93046deb01a6e57c1792cd1f6cf4f9d8a4b055ec627b2d9f336a3e3af439fbd708a934d933e8ef44e797bf279b5ba95b72cd0cacb95211298e748e69b93849cc564d9ddf2ef10567a638732eacee3eb960d4760f21adf09bf5088a11ef4dffce3a3dd31c4fcaa3169af889bca315db499173d6118ce8a892d115501d94702e0597ea80ce792907693544bf728970883b84c4b895558c090d94f6c84ad8fd5a8c2dcf2823b0b28cfcfb82feeb38514c14726e250b367efa1217d84ea63b51f171ec707cae8317675f3dd32ce99a91e2e6be526cab89269101ed97ed61f15243e47afab5bd72fdc6dc88d9127769e6fc9781e72b942729f45cbab57e5cfda8e0213663f0526a2088d8441728aa26b3941471b5ab23c90c294c38966d62568ff12e2a66ced08916a224468dfde73693befe8a1fd0f1081b33491127ce0a551b7c57c9e56ee6e4b02d83c10fb3074867dc31438d580d23931c8203aefbad34220c12021c606ead50e048d5c9802aca267160babd2e041e40180f248d5e43e3615922d61a37f9597f23b81b0fdbd57cb82ce8f195d33631dfbab8d2d70f708f0978b089febd76a2368443b80781329377d70d6fdbe9a9a5fe55ca080bf54e3e05a5b235cc7a7aa91afe7f84aabb50bf37ce854e48ff61b86fd56dbee5e85f38dc18e25e9bc2c8f0e9c42e8cd69c188e5c2ce7d612de05f07118e5b766b8ce74afefc9c834e678621952fe5504b940793eeb580e654e3f7810136db6f533166708355e6eee313352e583aefc0f1b22d6660b969d9965317bf193cdc2da4d870d3385c85873f99f17d7bfa156bd7913d811da2368b427701ff7e4c1cb3d53eb9f51a54366811618df679bd97e1b52837bfaf1725ddde736fdd1173bab5c17d477d7cbb0c143e898f1072f75f72638a7f8c7298d71851f47c674dd8b91f4447cd174649814f57aa99c45dd961be3a52a1754806f58fb2d619a11f8e0d4bd4d07ac35fd248baff4dc259105472755cd5aebcf99ebd3f908dbe97ad2567329995b8c47e856e9a7f1e678520975e6d837d44c555907a159d837c8981af3a3002db05e8c8a8218007f43a8474f672fb280df312b8d6c6f128822f9f7b4b09201bddc72e0b09477e984ac0a0e3cd740cb79ff30e69de909920578b87ef4f672fbd3582ed245a2fcf0db7e7ff682e05c2a70d043dac94d290a2e2d5d10d5100707a13bc27ae27d141bcbd12c58619aa7329e9b4f2ee978cb3a0b8e026b0f7c813f01f081e250af9675eb801082fd7f360a2cff7d83b0bbdee109af67ce03d52924ca683ae97e171ca0cbfab9417b3afc488168e440321893b2fdf823e096fa0af07d9d3db22d94bee9219f9e9cb9643b323f6b9abf6963f7f42023662beeb42e376dad3bc9212660dbe9aa93f036b3b62fc90049b6407e2c0d88d12da15c0511b8576926c239b537e9678010f1cdf110aadeb4c5ae1e52410accd4b1a0086785f98ac52e6b8cee557588f59ce7dd61785e61ea6697376d5e16771bfef2cb7c6eebc99fbdd04d565bf626ec956b6eaeb3a3a901f63d673f8c4e286cb4e40e0dc65d092abc6156118c7b727ab58cf4e8946fdc7857a8290337af23e47046462d0cfa064e0f3b6feb70b334d4142100c82669430aab289734c3f288f023ab492a0f4f3b2f4d298a7e6ca3ed1ab241f348b3435ce1e0ba5a8dd90d34a96ef944803ddbbfc0e80fcbc0df104f03f703";
```

Comment:
Can't we use one of the class definition fixtures we already have in `gateway-test-fixtures::zstd_compressed_contracts` here?

Reply:
`gateway-test-fixtures::zstd_compressed_contracts` just contains binary blobs. What's complicated about those? I was just suggesting using one of those instead of adding yet another class definition.

Reply:
My point is that thinks like `get_contract_class_from_fixture` are not necessary when single blob is enough. And for `similateTransaction` that single blob is enough, along with single dummy block.

Reply:
You don't have to use `get_contract_class_from_fixture`, it's not even in that crate. All the `gateway-test-fixtures` crate has is various blobs with contract definitions.

Reply:
My bad, I was looking at the wrong sub-crate. Still for this test case it is enough to go with single contract blob.

Reply:
For this specific test case I used a transaction I executed manually against testnet: https://testnet.starkscan.co/class/0x02b63cad399dd78efbc9938631e74079cbf19c9c08828e820e7606f46b947513

Reply:
I get that -- but you could pretty much have the same test which deploys one of the classes we already have as a fixture?

Something like this: https://github.com/eqlabs/pathfinder/commit/9697f36204c7e57ce0342712f961c64ef7d6a7d2

Reply:
Done: https://github.com/eqlabs/pathfinder/pull/977/commits/fda848df19080424c9ac5342c786237b0ab60363, https://github.com/eqlabs/pathfinder/pull/977/commits/92d1f2c0f18c4d68653db0a9cbc23b1e0e47498d